core.spec.md

```diff
@@ -0,0 +1,501 @@
+</dl>
+
+The version tag MUST be a valid {VersionTag}.
+
+#### Ignore meaningless URL components
+
+When extracting the URL's `name` and `version`, processors MUST ignore any url components which are not assigned a meaning. This spec assigns meaning to the final two segments of the [path](https://tools.ietf.org/html/rfc3986#section-3.3). Other URL components—particularly query strings and fragments, if present—MUST be ignored for the purposes of extracting the `name` and `version`.
+
+```html diagram -- Ignoring meaningless parts of a URL
+<code class=anatomy>
+  <span class=pink style='--depth: 2'>https://example.com/<span>exampleSpec<aside>name</aside></span><aside>identity</aside></span>/<span style='--depth: 2' class=green>v1.0<aside>version</aside></span><span class=grey>?key=val&k2=v2#frag<aside>ignored</aside></span>
+</code>
+```
+
+#### Why is versioning in the URL, not a directive argument?
+
+The version is in the URL because when a human reader visits the URL, we would like them to be taken to the documentation for the *version of the feature used by this document*. Many text editors will turn URLs into hyperlinks, and it's highly desirable that clicking the link takes the user to the correct version of the docs. Putting the version information in a separate argument to the {@core} directive would prevent this.
+
+###! as: String
+
+Change the [names](#sec-Prefixing) of directives and schema elements from this specification. The specified string MUST be a valid GraphQL identifier and MUST NOT contain the namespace separator (two underscores, {"__"}).
+
+When [`as:`](#@core/as) is provided, processors MUST replace the default name prefix on the names of all [prefixed schema elements](#sec-Elements-which-must-be-prefixed) with the specified name.
+
+```graphql example -- Using {@core}(feature:, as:) to use a feature with a custom name
+schema
+  @core(feature: "https://spec.example.com/core/v1.0")
+  @core(feature: "https://spec.example.com/example/v1.0", as: "eg")
+{
+  query: Query
+}
+
+type User {
+  # Specifying `as: "eg"` transforms @example into @eg
+  name: String @eg(data: ITEM)
+}
+
+# Additional specified schema elements must have their prefixes set
+# to the new name.
+#
+# This data enum was specified as `example__Data`, but will be renamed
+# as `eg__Data`:
+enum eg__Data {
+  ITEM
+}
+
+# Name transformation must also be applied to definitions pulled in from
+# specifications.
+directive @eg(data: eg__Data) on FIELD
+
+# (...other definitions omitted...)
+```
+
+###! export: Boolean! = false
+
+Control whether the entire feature is exported to the API.
+
+If {true}, all schema elements--including all directive usages--from this feature will be included in the schema's API *unless* they are annotated with {@core__export}`(export: false)`.
+
+If {false} (the default), no schema elements from this feature will be included in the schema's API *unless* they are annotated with {@core__export}.
```

Comment:
SUBSTANTIVE (and honestly a brand new kind of question I had not previously considered): "or `@core__export(export: true)" ? I guess it's sort of implicit because of the default value.

(This does raise the question of whether a Core Schema processor should validate that a given document doesn't accidentally declare `directive @core__export(export: Boolean! = false)`!!! Thoughts (post merge)? I don't think either current implementation looks at the declaration. Or alternatively, this spec should not specify the behavior of `@core__export` and what that does depends on whether the document declares `export` to have a default value and what that value is??? Same for the `export` argument to `@core`.)

Reply:
I feel like there are two consistent answers here:

(1) If a spec wants to have default values for directive arguments, processors should enforce that any provided default value matches the specified one.

(2) It's not a good idea for specs to specify default values for directive arguments. They should instead specify nullable arguments, and state in the text how null should be treated. Schema authors may put default arguments in their definitions and those values should be respected when directives are invoked without specifying the argument. (There's a bit of messiness about the difference between null and argument not given here.)